### PR TITLE
Fix #6242, Check nil for sock.read

### LINKED ISSUE #6242: sock.read needs to check NilClass
I notice that when using Rex::IO::Stream#read, some modules don't check nil, but should.

Anything that is using the rex socket and is using #read could hit this problem.

### PR DESCRIPTION
**Pull request by wchen-r7**
Fix #6242

The #read method returns nil:
https://github.com/rapid7/metasploit-framework/blob/master/lib/rex/io/stream.rb#L79

And I found these modules not checking it.
## Verification
- [x] Just code reading. Make sure the updated modules are checking nil while using #read

### CODE REVIEW

File: modules/auxiliary/scanner/db2/discovery.rb [hunk 1]
```diff
@@ -32,33 +32,37 @@ def run_host(ip)
...
+        :type   => 'SERVICE_INFO',
+        :data   => res[2] + "_" + res[1]
+        )
+
+      report_service(
+        :host => ip,
+        :port => datastore['RPORT'],
+        :proto => 'udp',
+        :name => "ibm-db2",
+        :info => res[2] + "_" + res[1]
```
Comment (jlee-r7) [inline modules/auxiliary/scanner/db2/discovery.rb]: If there were no `\x00` bytes, both of these indices can be `nil` as well.
```ruby
:info => "#{res[2]}_#{res[1]}"
```
is safer.

File: modules/auxiliary/scanner/sap/sap_router_info_request.rb [hunk 2]
```diff
@@ -109,7 +109,7 @@ def run_host(ip)
       print_good("#{host_port} - Connected to saprouter")
       print_good("#{host_port} - Sending ROUTER_ADM packet info request")
       sock.put(ni_packet)
-      packet_len = sock.read(4).unpack('H*')[0].to_i 16
+      packet_len = sock.read(4).to_s.unpack('H*')[0].to_i 16
```
Comment (jlee-r7) [inline modules/auxiliary/scanner/sap/sap_router_info_request.rb]: If this is `nil`, converting it to a String here just delays the problem. In fact, if I'm reading the `while` loop below, it results in an infinite loop.

File: modules/auxiliary/scanner/motorola/timbuktu_udp.rb [hunk 3]
```diff
@@ -52,8 +52,11 @@ def run_host(ip)
         else
           print_error("Unable to determine info for #{ip}...")
         end
+    rescue ::Errno::EPIPE, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionRefused => e
+      vprint_error(e.message)
+      elog("#{e.class} #{e.message}\n#{e.backtrace * "\n"}")
```
Comment (jlee-r7) [inline modules/auxiliary/scanner/motorola/timbuktu_udp.rb]: If all you're doing is printing and logging the error, you shouldn't bother `rescue`ing it in the first place. Just let the console handle it.

Comment (wchen-r7) [inline modules/auxiliary/scanner/motorola/timbuktu_udp.rb]: According to the report, the console doesn't print it though...

Comment (wchen-r7) [inline modules/auxiliary/scanner/motorola/timbuktu_udp.rb]: Never mind, I was thinking the other pull request. Sorry my bad.

**Comment by wchen-r7** [comment]
OK got them all, thanks @jlee-r7

**Comment by bcook-r7** [comment]
Looks good to me.

**Pull request merged**